pxtwinrt/serial.ts

```diff
@@ -1,51 +1,82 @@
 /// <reference path="./winrtrefs.d.ts"/>
 
+type DeviceWatcher = Windows.Devices.Enumeration.DeviceWatcher;
+type DeviceInfo = Windows.Devices.Enumeration.DeviceInformation;
+type SerialDevice = Windows.Devices.SerialCommunication.SerialDevice;
+
 namespace pxt.winrt {
-    let watcher: any;
-    let ports: pxt.Map<DevicePort> = {}
-    let options: pxt.AppSerial;
+    let watcher: DeviceWatcher;
+    let deviceNameFilter: RegExp;
+    let activePorts: pxt.Map<DevicePort> = {}
 
     interface DevicePort {
-        info: any;
-        device?: any;
+        info: DeviceInfo;
+        device?: SerialDevice;
     }
 
     export function initSerial() {
         if (!pxt.appTarget.serial
             || !pxt.appTarget.serial.log
             || !pxt.appTarget.serial.nameFilter) return;
 
-        const filter = new RegExp(pxt.appTarget.serial.nameFilter);
-        const serialDeviceSelector = (Windows.Devices as any).SerialCommunication.SerialDevice.getDeviceSelector();
+        deviceNameFilter = new RegExp(pxt.appTarget.serial.nameFilter);
+        const serialDeviceSelector = Windows.Devices.SerialCommunication.SerialDevice.getDeviceSelector();
```

Comment:
can we specify the productId/deviceId here?

Reply:
might be able to; will take a look on the v0 side and port that change back to master if it works